Update style guide with a line about links.

patrickvacek opened the pull request:
Derived from https://github.com/uptane/deployment-considerations/pull/38 and Lois's suggestion that my proposals should be uniformly adopted.

tkfu commented:
Couple of minor comments:

* We don't (currently) have "rendered and released" versions of the deployment considerations.
* We should make it clear that this is the recommendation for linking to sections of the standard _from outside the standard_; internal links within the document should use the standard cross-link syntax.

jhdalek55 commented:
These are both good clarifications, Jon. Your mention of the need for a "release" version of the Deployment pages brings something I want to bring up...a "Table of Contents" for the introduction page where perhaps sections or subsections on the different pages can be accessed quickly.

patrickvacek commented:
> * We don't (currently) have "rendered and released" versions of the deployment considerations.
> 
> * We should make it clear that this is the recommendation for linking to sections of the standard _from outside the standard_; internal links within the document should use the standard cross-link syntax.

Good points. I've addressed both issues and fixed some more Deployment Consideration links in the readme.
